src/script/descriptor.cpp

```diff
@@ -493,12 +508,12 @@ class BIP32PubkeyProvider final : public PubkeyProvider
 /** Base class for all Descriptor implementations. */
 class DescriptorImpl : public Descriptor
 {
-    //! Public key arguments for this descriptor (size 1 for PK, PKH, WPKH; any size for Multisig).
+protected:
+    //! Public key arguments for this descriptor (size 1 for PK, PKH, WPKH; any size for WSH and Multisig).
     const std::vector<std::unique_ptr<PubkeyProvider>> m_pubkey_args;
     //! The string name of the descriptor function.
     const std::string m_name;
```

Comment:
In e9cac52614d11e30d1f6fec33efce3d00953bd70 "Miniscript support in output descriptors"

nit: `m_name` does not need to be `protected`.

Reply:
Yeah, but 
```patch
diff --git a/src/script/descriptor.cpp b/src/script/descriptor.cpp
index 66c4e1cba9..b40e330f67 100644
--- a/src/script/descriptor.cpp
+++ b/src/script/descriptor.cpp
@@ -508,11 +508,12 @@ public:
 /** Base class for all Descriptor implementations. */
 class DescriptorImpl : public Descriptor
 {
+    //! The string name of the descriptor function.
+    const std::string m_name;
+
 protected:
     //! Public key arguments for this descriptor (size 1 for PK, PKH, WPKH; any size for WSH and Multisig).
     const std::vector<std::unique_ptr<PubkeyProvider>> m_pubkey_args;
-    //! The string name of the descriptor function.
-    const std::string m_name;
 
     //! The sub-descriptor arguments (empty for everything but SH and WSH).
     //! In doc/descriptors.m this is referred to as SCRIPT expressions sh(SCRIPT)
```

Would make me reorder the initializer list of all classes heriting from `DescriptorImpl`, so i guess it's not so bad if it's just `protected` instead of `private`? Happy to change if you prefer, still. (This or doing `protectred:` then `private:` then `protected:` again.)